src/Routes/PrivateRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { Route, Redirect, RouteProps } from "react-router-dom";
+
+export interface ProtectedRouteProps extends RouteProps {
+  isAuthenticated?: boolean;
+  isLoadingAuthState?: boolean;
+  authenticationPath: string;
+}
+
+export const ProtectedRoute: React.FC<ProtectedRouteProps> = props => {
+  if (props.isLoadingAuthState) {
+    return (
+      <div>
+        <h1>Loading...</h1>
+      </div>
+    );
+  }
+
+  let redirectPath = '';
+  if (!props.isAuthenticated) {
+    redirectPath = props.authenticationPath;
+  }
+
+  if (redirectPath) {
+    const renderComponent = () => <Redirect to={{ pathname: redirectPath }} />;
+    return <Route {...props} component={renderComponent} render={undefined} />;
+  } else {
+    return <Route {...props} />;
+  }
```

Comment:
```suggestion
  if (!props.isAuthenticated) {
    const pathname = props.authenticationPath;
    const redirect = () => <Redirect to={{ pathname }} />;
    return <Route {...props} component={redirect}  />;
  }
  
  return <Route {...props} />;
```

The `render={undefined}` is strange. I guess it is needed if you wrote it that way.

Reply:
Your variation looks much better, thanks.